props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,720 @@
+Clip out                  - Clip Detect pad Latched Off.
+Unjam                     - Bang the blaster or Reload.
+
+- If there's a dedicated POWER button,
+Power On / Off            - Click POW. - TODO: this with dual_prop is....weird
+- If there's a dedicated Reload button, 
+Reload                    - Click RELOAD.
+
+Wavs to use for talking Mode (else Talkie voice):
+- If these are not present, mode.wav will be used for all modes.
+- If no mode.wav either, then Talkie voice speaks selected mode.
+  mdstun.wav
+  mdkill.wav
+  mdauto.wav
+Additionally:
+  quote.wav      // for playing quotes
+  in.wav         // for blaster power off
+  out.wav        // for blaster power on
+  destruct.wav   // for self destruct overload
+  boom.wav       // for self destruct explosion
+These can be used to match a more "plasmatic" stun sound.
+  clipins.wav
+  clipouts.wav
+  reloads.wav
+
+The following are likely best kept in a 'common' folder in your Font Search Path, but of course
+can be specific versions within a font as well.
+Choose a Voice Pack containing all these sounds here: https://fredrik.hubbe.net/lightsaber/sound/
+
+  Battery level wavs:
+    mzero.wav
+    mnum 1-20 wavs, as well as the "tens" thirty.wav - ninety.wav
+    battery.wav    // for accompanying EFFECT_BATTERY_LEVEL
+    battlevl.wav   // for Spoken Battery Level
+    mpercent.wav
+    mpoint.wav
+    mvolts.wav
+  vmbegin.wav    // for Begin Volume Menu
+  vmend.wav      // for End Volume Menu
+  volup.wav      // for increse volume
+  voldown.wav    // for decrease volume
+  volmin.wav     // for minimum volume reached
+  volmax.wav     // for maximum volume reached
+  mfalse.wav     // "disabled" used for sequential quote mode
+  mtrue.wav      // "enabled" used for sequential quote mode
+  mrandom.wav    // "ramdom" used for sequential quote mode
+*/
+
+#ifndef PROPS_BLASTER_BC_BUTTONS_H
+#define PROPS_BLASTER_BC_BUTTONS_H
+
+#ifndef BLASTER_DEFAULT_MODE
+#define BLASTER_DEFAULT_MODE MODE_KILL
+#endif
+
+#include "blaster.h"
+#include "../sound/sound_library.h"
+
+#ifdef PROP_TYPE
+#undef PROP_TYPE
+#endif
+
+#define PROP_TYPE BlasterBCButtons
+
+
+// Additional effects for BC buttons prop.
+EFFECT(clipins);    // s for stun mode
+EFFECT(clipouts);   // s for stun mode
+EFFECT(reloads);    // s for stun mode
+EFFECT(destruct);
+#ifndef PROPS_DUAL_PROP_H
+EFFECT(battery);    // for EFFECT_BATTERY_LEVEL
+EFFECT(vmbegin);    // for Begin Volume Menu
+EFFECT(vmend);      // for End Volume Menu
+EFFECT(volup);      // for increse volume
+EFFECT(voldown);    // for decrease volume
+EFFECT(volmin);     // for minimum volume reached
+EFFECT(volmax);     // for maximum volume reached
+#endif
+
+class BlasterBCButtons : public Blaster {
+public:
+  BlasterBCButtons() : Blaster() {}
+  const char* name() override { return "BlasterBCButtons"; }
+
+  void NextBlasterMode() override {
+    switch(blaster_mode) {
+      case MODE_STUN:
+        stun_mode_ = false;
```

Comment:
stun_mode_ seems redundant, why not just check if blaster_mode == MODE_STUN ?

Reply:
I think this is to play dedicated stun reload and clip in/out sounds. It's used as a condition in SB_Effect.

Reply:
removed, used blaster_mode == MODE_STUN instead in the conditional checks.

Reply:
done